File: Data Structures & Algorithms/house-robber-ii/submission-3.cs
public class Solution {
    public int Rob(int[] nums) {
        int a = 0, b = nums[0];
        for(int i=1;i<nums.Length-1;i++){
            int rob = Math.Max(nums[i]+a, b);
            a = b;
            b = rob;
        }
        if(nums.Length==1) return b;
        int c = 0, d = nums[1];
        for(int i=2;i<nums.Length;i++){
            int rob = Math.Max(nums[i]+c, d);
            c = d;
            d = rob;
        }
        return Math.Max(b,d);
    }
}

File: Data Structures & Algorithms/house-robber/submission-3.java
class Solution {
    public int rob(int[] nums) {
        int r1 = 0;
        int r2 = 0;
        for(int num:nums){
            int tmp = r1;
            r1 = r2;
            r2 = Math.max(tmp+num,r2);
        }
        return Math.max(r1,r2);
    }
}
